src/cleaner.ts

```
@@ -1,390 +1,535 @@
+	verbose: boolean;
+
+	/**
+	 * Create a new GitLabContainerRepositoryCleaner instance,
+	 *
+	 * @param options
+	 */
+	constructor(options: Partial<GitLabContainerRepositoryCleanerOptions> = {}) {
+		this.gl = new Gitlab({
+			token: options.gitlabToken || process.env.GITLAB_TOKEN || "",
+			host: options.gitlabHost || process.env.GITLAB_HOST || "",
+		});
+
+		this.dryRun = options.dryRun ?? false;
+		this.concurrency = options.concurrency ?? DEFAULT_CONCURRENCY;
+		this.verbose = options.verbose ?? false;
+	}
+
+	/**
+	 * Get Container Repositories in a range of ID. Look for repository for each ID in range concurrently using GitLab API:
+	 * a 404 indicated repository does not exists, otherwise repository data is returned.
+	 * @param startIndex repository ID to start from
+	 * @param endIndex repository ID to end by
+	 * @param outputFile path to write results as JSON
+	 */
+	public async getContainerRepositoriesConcurrently(
+		startIndex = DEFAULT_START_INDEX,
+		endIndex = DEFAULT_END_INDEX,
+		outputFile: string | undefined = undefined,
+	): Promise<RegistryRepositorySchema[]> {
+		if (!outputFile || outputFile === "") {
+			console.log(
+				"You didn't specify an output path to write results. By default results will be shown on stdout.",
+			);
+			console.log(
+				"Output may be long, it's possible your console buffer won't show everything.",
+			);
+			console.log(
+				"This command may run for a long time and some results may be lost.",
+			);
+			console.log(
+				"Use -o flag to specify a file such as -o /tmp/repositories.json",
+			);
+			console.log("");
+
+			if (stdin.isTTY && process.env.NODE_ENV !== "test") {
+				await this.promptUser(
+					"Press CTRL+C to interrupt or ENTER to continue...",
+				);
+			} else {
+				console.log(
+					"Non-interactive environment detected. Continuing without prompt.",
+				);
+			}
+		}
+
+		if (startIndex > endIndex) {
+			console.error("Start index is greater than end index");
+			process.exit(1);
```

Maybe throw an exception here instead ?